### ISSUE TITLE
Self-lint with eslint-plugin-eslint-plugin (recommended)

### ISSUE THREAD

**Issue by rakleed** [gajus/eslint-plugin-jsdoc#1776]
## Motivation
`eslint-plugin-jsdoc` is an ESLint plugin, but its own eslint config doesn't appear to lint the plugin's rule files with [`eslint-plugin-eslint-plugin`](https://github.com/eslint-community/eslint-plugin-eslint-plugin). Its `recommended` config catches things that are easy to miss by hand and silently break for end users — most notably `no-deprecated-context-methods` / `no-deprecated-report-api`, which flag calls like `context.getSourceCode()` that ESLint has been removing across major versions, so a rule keeps working until an ESLint upgrade crashes it at runtime with no prior warning.

## Current behavior
The eslint config here doesn't apply `eslint-plugin-eslint-plugin`'s `recommended` config to the rule files at all, so none of that class of issue is caught before it ships.

## Desired behavior
Add `eslint-plugin-eslint-plugin` as a devDependency and apply its `recommended` config to the rules directory:

```js
import eslintPlugin from 'eslint-plugin-eslint-plugin';

export default [
  // ...existing config
  {
    ...eslintPlugin.configs.recommended,
    files: ['**/rules/**/*.js'], // adjust to your rules path
  },
];
```

## Alternatives considered
Manual review catches some of this, but it's exactly the kind of mechanical check a linter is better suited for. Several other rule-author repos already run this exact config (eslint-stylistic, eslint-plugin-import-x, eslint-plugin-perfectionist, eslint-plugin-unicorn, eslint-plugin-jsx-a11y-x, facebook/react's own eslint-plugin-react-hooks), so it's a well-trodden setup rather than a novel ask.
